backend/src/main/resources/org/cryptomator/hub/flyway/V4__Update_PKs.sql

```diff
@@ -0,0 +1,50 @@
+-- noinspection SqlNoDataSourceInspectionForFile
+
+-- temporarily drop views
+DROP VIEW "effective_vault_access";
+DROP VIEW "effective_group_membership";
+
+-- temporarily disable foreign key constraints
+ALTER TABLE "vault_access" DROP CONSTRAINT "VAULT_ACCESS_FK_VAULT";
+ALTER TABLE "access_token" DROP CONSTRAINT "ACCESS_FK_VAULT";
+ALTER TABLE "access_token" DROP CONSTRAINT "ACCESS_FK_DEVICE";
+
+-- collate C is more efficient on indexed columns (sadly user ids are generated externally and there is no guarantee for them to be UUIDs):
+ALTER TABLE "authority" ALTER COLUMN "id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "group_membership" ALTER COLUMN "group_id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "group_membership" ALTER COLUMN "member_id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "user_details" ALTER COLUMN "id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "group_details" ALTER COLUMN "id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "vault_access" ALTER COLUMN "authority_id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "device" ALTER COLUMN "owner_id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "device" ALTER COLUMN "id" SET DATA TYPE VARCHAR(64) COLLATE "C";
```

Comment:
Why not use UUID for devices as well?

Reply:
Existing entries are not UUIDs. Instead we have hex-encoded sha256 fingerprints of the device's public key.